### PR TITLE
Contract results endpoint should include evm_address field for created contracts

### LINKED ISSUE #4841: Contract results endpoint should include `evm_address` field for created contracts
The EVM address of contracts must be used in some scenarios or the consensus node errors out with `INVALID_ACCOUNT_ID`. In order to avoid having to make additional calls to check whether a newly created contract also has an EVM address, we should update the contract results endpoint to include an `evm_address` field for created contracts.

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #4841

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — COMMENTED** [review]
Mostly good.
Curious about the null case though

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 1]
```diff
@@ -1976,6 +1976,11 @@ components:
               nullable: true
               type: string
               example: '0x0127'
+            evm_address:
+              description: The hex encoded evm address
+              example: "0x25fe26adc577cc89172e6156c9e24f7b9751b762"
+              nullable: true
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply hedera-mirror-rest/api/v1/openapi.yml:1982]: don't think it's actually `nullable`, the field is either in the response with a non null value or not in the response. In addition, I'm not sure I agree that the field is populated conditionally depending on the endpoints.

File: hedera-mirror-rest/__tests__/specs/contracts/results/{id}/no-evm-address.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    "amount": 2000000000,
+    "block_gas_used": 50000000,
+    "block_hash": "0x6ceecd8bb224da491",
+    "block_number": 17,
+    "bloom": "0x0505",
+    "call_result": "0x0606",
+    "chain_id": "0x",
+    "contract_id": "0.0.5001",
+    "created_contract_ids": ["0.0.7001"],
+    "error_message": null,
```
Comment (xin-hedera) [inline hedera-mirror-rest/__tests__/specs/contracts/results/{id}/no-evm-address.json:87]: the response should show `"evm_address": "0x"`

**Review by xin-hedera — CHANGES_REQUESTED** [review]

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — DISMISSED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LG

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**